application/models/SurveysGroups.php

```diff
@@ -197,7 +197,7 @@ public function search()
             'criteria'=>$criteria,
         ));
 
-        $dataProvider->setTotalItemCount(count($dataProvider->getData()));
+        $dataProvider->setTotalItemCount($this->count($criteria));
```

Comment:
Ehm? $criteria is just an object, no?

Reply:
$this is the current model : SureysGroup::model()->count($criteria)

Reply:
Oh ok